### PR TITLE
UI2 Create Account choose username

### PR DESCRIPTION
**Pull request by dbo1dyrev**
**PR Requirements**

If you have made any visual changes to the GUI. Make sure you have:

- [ ]  Tested on iOS Tablet
- [x] Tested on small Android
- [ ] n/a

Screenshot 2021-05-31 at 13 09 06

### CODE REVIEW

File: packages/edge-login-ui-rn/src/reducers/CreateUserReducer.js [hunk 1]
```diff
@@ -15,7 +15,8 @@ export type CreateState = {|
   +pinError: string,
   +pinErrorMessage: string | null,
   +username: string | null,
-  +usernameErrorMessage: string | null
+  +usernameErrorMessage: string | null,
+  +checkUsernameAvailabiltyPending: boolean
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/reducers/CreateUserReducer.js:19]: This should be local scene state. We are trying to reduce the amount of stuff in Redux.

File: packages/edge-login-ui-rn/src/components/themed/EdgeTextField.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
+// @flow
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/components/themed/EdgeTextField.js:1]: Didn't Andrew create a new text field component specifically for Edge? We should be using his new work instead of react-native-material-textfield.

File: packages/edge-login-ui-rn/src/components/modals/TextInputModal.js [hunk 3]
```diff
@@ -103,7 +103,7 @@ export class TextInputModal extends React.Component<Props, State> {
       <ThemedModal bridge={bridge} onCancel={this.handleCancel}>
         {title != null ? <TitleText>{title}</TitleText> : null}
         {message != null ? <MessageText>{message}</MessageText> : null}
-        <ThemedTextField
+        <EdgeTextField
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/components/modals/TextInputModal.js:106]: Updating the modal to match our style is a good idea, but it should be its own commit.

File: packages/edge-login-ui-rn/src/util/hooks.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// @flow
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/util/hooks.js:1]: Enabling hooks is good, but this should be its own commit.

File: packages/edge-login-ui-rn/src/components/common/MaterialInput.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // @flow
 
 import * as React from 'react'
-import { TextField } from 'react-native-material-textfield'
+import { TextField } from 'react-native-material-textfield-legacy'
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/components/common/MaterialInput.js:4]: Why was this changed? Why was the new package added as a dependency in the first place?

File: packages/edge-login-ui-rn/src/common/locales/strings/de.json [hunk 6]
```diff
@@ -56,7 +56,6 @@
   "terms_two": "Ich verstehe, dass wenn ich dieses Gerät verliere, oder die App deinstalliere, mein digitales Vermögen nur noch mit Benutzernamen und Passwort wiederhergestellt werden können",
   "terms_three": "Ich verstehe, dass wenn ich meinen Benutzernamen und mein Passwort verliere, Edge mein Benutzerkonto nicht wiederherstellen kann, sollte vorher nicht die Passwort-Wiederherstellung aktiviert worden sein",
   "terms_four": "Ich verstehe, dass ich für die Sicherstellung meiner Passwörter, privaten Schlüsselpaare, PIN und alle anderen Codes für den Zugriff auf die Software verantwortlich bin. %s ist nicht verantwortlich, wenn meine Informationen kompromittiert werden oder von Dritten aufgerufen werden, bei denen das Geld verloren geht",
-  "step_one": "Schritt 1/3",
```
Comment (swansontec) [inline packages/edge-login-ui-rn/src/common/locales/strings/de.json:59]: When editing strings, please don't touch the translations. We have automated tools to manage this.

File: packages/edge-login-ui-rn/src/components/common/MaterialInput.js (hunk 5 above)
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/components/common/MaterialInput.js:4]: Because previous "react-native-material-textfield" is an old fork with additional commit but I need to use more fresh version of original repo. I left it for an old code that continue to use this fork. After updating the entire UI to a new one, it will be removed.

File: packages/edge-login-ui-rn/src/components/themed/EdgeTextField.js (hunk 2 above)
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/components/themed/EdgeTextField.js:1]: It's not merged yet

File: packages/edge-login-ui-rn/src/components/modals/TextInputModal.js (hunk 3 above)
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/components/modals/TextInputModal.js:106]: Return this back with old component

**Comment by dbo1dyrev** [comment]
*with new EdgeTextFieldOutlined
Screenshot 2021-06-14 at 16 37 51

File: packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js [hunk 7]
```diff
@@ -69,7 +69,7 @@ export function PrimaryButton(props: Props) {
       onPress={onPress}
     >
       {label != null ? <Text style={styles.primaryText}>{label}</Text> : null}
-      {spinner != null ? (
```
[the remark this answers is not in the document]
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js]: but this current solution is really strange. I need something like this
```
<SecondaryButton spinner={isLoading} />
```
where `isLoading` is boolean.
But here to show the spinner I need to pass undefined or null. Whaaat?

**Comment by paullinator** [comment]
Left margins look inconsistent. Please fix via rebase

**Comment by paullinator** [comment]
Right margins look inconsistent as well

File: packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (swansontec) [reply packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js:72]: If a value is a boolean, (`true` or `false`), you can use it the normal way. So, `<SecondaryButton spinner={isLoading} />` is great! But, if a value is `SomeThing | undefined`, you need to do the `!= null` check.

**Comment by dbo1dyrev** [comment]
Screenshot 2021-06-17 at 15 40 33

>Left margins look inconsistent. Please fix via rebase
>Right margins look inconsistent as well
fixed it

File: packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/components/themed/ThemedButtons.js:72]: something like `spinner === true` will be good?
or, maybe you can provide your short solution?
because using just `spinner != null` confusing when you trying pass the boolean false value and see spinner on button

File: packages/edge-login-ui-rn/src/actions/CreateAccountActions.js [hunk 8]
```diff
@@ -34,31 +34,36 @@ export function checkUsernameForAvailabilty(data: string) {
     const context = imports.context
     // dispatch(openLoading()) Legacy dealt with state for showing a spinner
     // the timeout is a hack until we put in interaction manager.
-    setTimeout(() => {
-      context
-        .usernameAvailable(data)
-        .then(async response => {
-          if (response) {
+    return new Promise<void>(resolve =>
+      setTimeout(() => {
```
Comment (thehobbit85) [inline packages/edge-login-ui-rn/src/actions/CreateAccountActions.js:38]: Are you sure the "setTimeout" is still needed with your changes?

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
See comment

File: packages/edge-login-ui-rn/src/actions/CreateAccountActions.js (hunk 8 above)
Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/actions/CreateAccountActions.js:38]: I just improve this action to return a promise, I think I don't resolve the comment above, but I can try to remove it and see what happens next

Comment (dbo1dyrev) [reply packages/edge-login-ui-rn/src/actions/CreateAccountActions.js]: ok, works fine, removed it

**Review by thehobbit85 — APPROVED** [review]

**Review by thehobbit85 — CHANGES_REQUESTED** [review]
Changes requested

**Comment by dbo1dyrev** [comment]
Add commits:
"Update Themed Buttons font styles"
"Update Secondary Button styles: change padding size and make label bold"
squash all fixup commits

Screenshot 2021-06-30 at 21 48 59

**Review by thehobbit85 — APPROVED** [review]
